src/modules/sync/index.tsx

```diff
@@ -136,27 +135,52 @@ export function ContentTabSync( { selectedSite }: { selectedSite: SiteDetails }
 		localSiteId: selectedSite.id,
 		userId: user?.id,
 	} );
-	const {
-		syncSites,
-		isFetching: isFetchingSyncSites,
-		refetchSites,
-	} = useFetchWpComSites( connectedSites.map( ( { id } ) => id ) );
 	const [ connectSite ] = useConnectSiteMutation();
 	const [ disconnectSite ] = useDisconnectSiteMutation();
 	const { pushSite, pullSite, isAnySitePulling, isAnySitePushing } = useSyncSites();
-	const isAnySiteSyncing = isAnySitePulling || isAnySitePushing;
-	const { streamlineOnboarding } = useFeatureFlags();
 
-	const [ selectedRemoteSite, setSelectedRemoteSite ] = useState< SyncSite | null >( null );
-	const [ pendingModalMode, setPendingModalMode ] = useState< 'push' | 'pull' | null >( null );
+	const connectedSiteIds = connectedSites.map( ( { id } ) => id );
+	const {
+		data: syncSites = [],
+		isLoading: isLoadingSyncSites,
+		isFetching: isFetchingSyncSites,
+		isSuccess: isSuccessSyncSites,
+		isUninitialized: isUninitializedSyncSites,
+		refetch: refetchWpComSites,
+	} = useGetWpComSitesQuery(
+		{ connectedSiteIds, userId: user?.id },
+		{ refetchOnMountOrArgChange: true }
+	);
 
-	// Open modal after fetch completes and state updates
+	const refetchSites = useCallback( async (): Promise< SyncSite[] > => {
+		if ( isUninitializedSyncSites ) {
+			return [];
+		}
+		try {
+			const result = await refetchWpComSites();
+			return result.data ?? [];
+		} catch ( error ) {
+			// Query might not be ready to refetch yet (e.g., was skipped due to offline)
+			console.warn( 'Failed to refetch sites:', error );
+			return [];
+		}
+	}, [ refetchWpComSites, isUninitializedSyncSites ] );
+
+	// Refetch sites when modal opens to check for newly created sites
 	useEffect( () => {
-		if ( pendingModalMode && ! isFetchingSyncSites ) {
-			dispatch( connectedSitesActions.openModal( pendingModalMode ) );
-			setPendingModalMode( null );
+		if ( isModalOpen && isAuthenticated && ! isUninitializedSyncSites ) {
+			refetchWpComSites().catch( ( error ) => {
```

Comment:
Would be possible to refetch wpcom sites inside openModal action or call it at the same time?